pkg/machines/components/networks/createNetworkDialog.jsx

```diff
@@ -0,0 +1,539 @@
+            else if (!utils.validateIpv6(dialogValues.ipv6DhcpRangeEnd))
+                validationFailed['ipv6DhcpRangeEnd'] = _("Invalid IPv6 address");
+            else if (!utils.isIpv6InNetwork(dialogValues.ipv6, dialogValues.prefix, dialogValues.ipv6DhcpRangeEnd))
+                validationFailed['ipv6DhcpRangeEnd'] = _("Address not within subnet");
+        }
+    }
+
+    return validationFailed;
+}
+
+const NetworkConnectionRow = ({ onValueChanged, dialogValues, loggedUser }) => {
+    return (
+        <React.Fragment>
+            <label className='control-label' htmlFor='create-network-connection'>
+                {_("Connection")}
+            </label>
+            <MachinesConnectionSelector id='create-network-connection'
+                dialogValues={dialogValues}
+                onValueChanged={onValueChanged}
+                loggedUser={loggedUser} />
+        </React.Fragment>
+    );
+};
+
+const NetworkNameRow = ({ onValueChanged, dialogValues, validationFailed }) => {
+    const validationState = validationFailed.name ? 'error' : undefined;
+
+    return (
+        <React.Fragment>
+            <label className='control-label' htmlFor='create-network-name'>
+                {_("Name")}
+            </label>
+            <FormGroup validationState={validationState} controlId='name'>
+                <input
+                   id='create-network-name'
+                   type='text'
+                   placeholder={_("Unique Network Name")}
+                   value={dialogValues.name}
+                   onChange={e => onValueChanged('name', e.target.value)}
+                   className='form-control' />
+                { validationState == 'error' &&
+                <HelpBlock>
+                    <p className='text-danger'>{validationFailed.name}</p>
+                </HelpBlock> }
+            </FormGroup>
+        </React.Fragment>
+    );
+};
+
+const NetworkForwardModeRow = ({ onValueChanged, dialogValues }) => {
+    const forwardModes = ['nat', 'open', 'none'];
+
+    return (
+        <React.Fragment>
+            <label className='control-label' htmlFor='create-network-forward-mode'>
+                {_("Forward Mode")}
+            </label>
+            <Select.Select id='create-network-forward-mode'
+                           initial={dialogValues.forwardMode}
+                           onChange={value => onValueChanged('forwardMode', value)}>
+                { forwardModes.map(mode => {
+                    return (
+                        <Select.SelectEntry data={mode} key={mode}>
+                            {rephraseUI('networkForward', mode)}
+                        </Select.SelectEntry>
+                    );
+                })
+                }
+            </Select.Select>
+        </React.Fragment>
+    );
+};
+
+const NetworkDeviceRow = ({ devices, onValueChanged, dialogValues, validationFailed }) => {
```

Comment:
validationFailed is not used. I don't know why React doesn't complain !? but it should be removed from the properties.